### PR TITLE
feat(bedrock): add guardrail support for agents

### LINKED ISSUE #561: (Bedrock agent): guardrail support
### Describe the feature

Guardrail support for agents has been added to the CDK recently: https://github.com/aws/aws-cdk/pull/30722 
https://github.com/aws/aws-cdk/releases/tag/v2.148.0

### Use Case

Missing feature

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by krokoko**
Fixes #561 

- Add guardrail support for agents
- Update CDK library version
- Update Projen version

Testing:
- Using the [https://github.com/aws-samples/generative-ai-cdk-constructs-samples/tree/main/samples/bedrock-agent](Bedrock Agent sample) , creating a guardrail and associating it with the agent

With ```agent.addGuardrail(guardrail);``` API:

image

Through constructor : 

```
const agent = new bedrock.Agent(this, 'Agent', {
      foundationModel: bedrock.BedrockFoundationModel.ANTHROPIC_CLAUDE_INSTANT_V1_2,
      instruction: 'You are a helpful and friendly agent that answers questions about literature.',
      knowledgeBases: [kb],
      enableUserInput: true,
      shouldPrepareAgent:true,
      guardrailConfiguration: {
        guardrailId: guardrail.guardrailId,
        guardrailVersion: guardrail.guardrailVersion
      },
    });
```

image

Note: when providing guardrail config through the constructor, the customer has to provide correct permissions to the agent to use the guardrail. This is because we are currently not able to retrieve programmatically a guardrail based on id/version. 

Use:

When asking a question, the trace shows correct use of guardrails:

image

image

When asking a question which falls under the guardrails boundaries, the correct fallback message is returned:

image

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

**Review by dineshSajwan — APPROVED** [review]

File: .projenrc.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ const project = new awscdk.AwsCdkConstructLibrary({
   authorOrganization: true,
   description: 'AWS Generative AI CDK Constructs is a library for well-architected generative AI patterns.',
   cdkVersion: CDK_VERSION,
-  projenVersion: '~0.78.8',
+  projenVersion: '~0.80.10',
```
Comment (scottschreckengaust) [inline .projenrc.ts:46]: adds a helpful needs in the build:

```suggestion
  projenVersion: '~0.84.5',
```

File: .projenrc.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ const project = new awscdk.AwsCdkConstructLibrary({
...
   constructsVersion: '10.3.0',
   defaultReleaseBranch: 'main',
   jsiiVersion: '~5.3.0',
```
Comment (scottschreckengaust) [inline .projenrc.ts:49]: `~5.4.0`

File: .projenrc.ts [hunk 3]
```diff
@@ -158,6 +158,7 @@ project.github?.actions.set('actions/stale@v4', 'actions/stale@a20b814fb01b71def
 project.github?.actions.set('actions/upload-artifact@v3', 'actions/upload-artifact@18bf333cd2249fbbbdb605fd9d9ed57efd7adf34'); // https://github.com/projen/projen/issues/3529
 project.github?.actions.set('actions/upload-artifact@v4', 'actions/upload-artifact@18bf333cd2249fbbbdb605fd9d9ed57efd7adf34');
 project.github?.actions.set('amannn/action-semantic-pull-request@v5.0.2', 'amannn/action-semantic-pull-request@01d5fd8a8ebb9aafe902c40c53f0f4744f7381eb');
+project.github?.actions.set('amannn/action-semantic-pull-request@v5.4.0', 'amannn/action-semantic-pull-request@e9fabac35e210fea40ca5b14c0da95a099eff26f');
```
Comment (scottschreckengaust) [inline .projenrc.ts:161]: use the same hash for other version too

File: .projenrc.ts (hunk 2 above)
Comment (scottschreckengaust) [inline .projenrc.ts:49]:
```suggestion
  jsiiVersion: '~5.4.0',
```

Comment (krokoko) [reply .projenrc.ts]: Done in [6abcd5c](https://github.com/awslabs/generative-ai-cdk-constructs/pull/564/commits/6abcd5c121c6e67f6308b7bd9bcbb1a557161876)

File: .projenrc.ts (hunk 1 above)
Comment (krokoko) [reply .projenrc.ts]: Done in [6abcd5c](https://github.com/awslabs/generative-ai-cdk-constructs/pull/564/commits/6abcd5c121c6e67f6308b7bd9bcbb1a557161876)

**Review by scottschreckengaust — APPROVED** [review]

**Review by dineshSajwan — APPROVED** [review]
LGTM

**Pull request merged**